tests/integration/targets/mso_schema_template_deploy_status/tasks/main.yml

```diff
@@ -0,0 +1,417 @@
+  mso_schema_template_deploy_status:
+    <<: *mso_info
+    schema: '{{ mso_schema | default("ansible_test") }}'
+    template: Template 1
+    state: status
+  register: status4_temp1
+
+- name: Verify status after deleting VRF1
+  assert:
+    that:
+    - status4_temp1 is changed
+    - status4_temp1.current.bds[0].state == 'deleted'
+    - status4_temp1.current.vrfs[0].state == 'deleted'
+
+- name: Deploy templates
+  mso_schema_template_deploy:
+    <<: *mso_info
+    schema: '{{ mso_schema | default("ansible_test") }}'
+    template: "{{ item }}"
+    site: '{{ mso_site | default("ansible_test") }}'
+    state: deploy
+  loop:
+  - Template 1
+  - Template 2
+
+- name: Check deployment status of Template 1
+  mso_schema_template_deploy_status:
+    <<: *mso_info
+    schema: '{{ mso_schema | default("ansible_test") }}' 
+    template: Template 1
+    state: status
+  register: status5_temp1
+
+- name: Check deployment status of Template 2
+  mso_schema_template_deploy_status:
+    <<: *mso_info
+    schema: '{{ mso_schema | default("ansible_test") }}'
+    template: Template 2
+    state: status
+  register: status5_temp2
+    
+- name: Verify status after deploying Templates to site
+  assert:
+    that:
+    - status5_temp1 is changed
+    - status5_temp2 is changed
+    - status5_temp1.current.anps == []
+    - status5_temp1.current.bds == []
+    - status5_temp1.current.vrfs == []
+    - status5_temp2.current.anps == []
+    - status5_temp2.current.bds == []
+    - status5_temp2.current.vrfs == []
+
+- name: Check status of all templates
+  mso_schema_template_deploy_status:
+    <<: *mso_info
+    schema: '{{ mso_schema | default("ansible_test") }}'
+    state: status
+  register: all_templates
+
+- name: Verify all
+  assert:
+    that:
+    - all_templates.current.policyStates | length == 2
+
+- name: Check Non-existing schema
+  mso_schema_template_deploy_status:
+    <<: *mso_info
+    schema: non-existing-schema
+    state: status
+  ignore_errors: yes
+  register: non_schema
+
+- name: Verify non_existing_schema
+  assert:
+    that:
+    - non_schema.msg == "Schema 'non-existing-schema' not found."
+
+- name: Check deployment status of non-existing-template
+  mso_schema_template_deploy_status:
+    <<: *mso_info
+    schema: '{{ mso_schema | default("ansible_test") }}'
+    template: non-existing-template
+    state: status
+  ignore_errors: yes
+  register: non_temp
+
+- name: Verify non_existing_template
+  assert:
+    that:
+    - non_temp.msg == "Template 'non-existing-template' not found."
+
+- name: Undeploy templates
+  mso_schema_template_deploy:
+    <<: *mso_info
+    schema: '{{ mso_schema | default("ansible_test") }}'
+    template: "{{ item }}"
+    site: '{{ mso_site | default("ansible_test") }}'
+    state: undeploy
+  loop:
+  - Template 1
+  - Template 2
+
+- name: Remove schemas
+  mso_schema:
+    <<: *mso_info
+    schema: '{{ mso_schema | default("ansible_test") }}'
+    state: absent
```

Comment:
I would prefer we move this on the next module test file as it might not be executed here when this test crash.

Reply:
Done